air/src/execution_step/lambda_applier/applier.rs

```diff
@@ -81,14 +122,35 @@ pub(crate) fn select_from_scalar<'value, 'accessor, 'i>(
         }
     }
 
-    Ok(value)
+    Ok(Cow::Borrowed(value))
+}
+
+fn select_by_functor_from_scalar(value: &JValue, functor: &Functor) -> ExecutionResult<JValue> {
+    match functor {
+        Functor::Length => {
+            let length = value
+                .as_array()
+                .ok_or_else(|| {
+                    ExecutionError::Catchable(Rc::new(CatchableError::LengthFunctorAppliedToNotArray(value.clone())))
+                })?
+                .len();
+            Ok(serde_json::json!(length))
+        }
+    }
 }
 
 impl<'value> StreamSelectResult<'value> {
-    pub(self) fn new(result: &'value JValue, tetraplet_idx: u32) -> Self {
+    pub(self) fn from_cow(result: Cow<'value, JValue>, tetraplet_idx: u32) -> Self {
         Self {
             result,
-            tetraplet_idx: tetraplet_idx as usize,
+            tetraplet_idx: Some(tetraplet_idx as usize),
+        }
+    }
+
+    pub(self) fn from_owned(result: JValue) -> Self {
```

Comment:
I believe that "owned" is misleading here.  It's not clear how it differs from `Self::from_cow(Cow::Owned(...))`.

`from_value`?  `from_new_value`? etc.